### PR TITLE
Ticketing system new

### PR DESCRIPTION
**Pull request by Bharani0012**
`Updated` Response and API endpoints for comments 

API Documentation;

https://sarantoontasticenterprise.atlassian.net/wiki/x/AYA7AQ

### CODE REVIEW

File: src/backend/db/database_manager.py [hunk 1]
```diff
@@ -850,5 +866,363 @@ def delete_ticket(self, ticket_id, customer_guid):
...
+                ORDER BY created_at ASC
+                LIMIT :page_size OFFSET :offset
+            """
+            results = session.execute(
+                text(query),
+                {"ticket_id": ticket_id, "page_size": page_size, "offset": offset},
+            ).fetchall()
+
+            comments_list = [
+                {"comment_id": row.comment_id, "comment": row.comment, "created_at": row.created_at}
```
Comment (SaravanakumarR2018) [inline src/backend/db/database_manager.py:1055]: The output should have
            CREATE TABLE IF NOT EXISTS ticket_comments (
                comment_id BIGINT AUTO_INCREMENT PRIMARY KEY,
                ticket_id BIGINT NOT NULL,
                posted_by VARCHAR(255) NOT NULL,
                comment TEXT NOT NULL,
                is_edited BOOLEAN DEFAULT FALSE,
                created_at TIMESTAMP DEFAULT CURRENT_TIMESTAMP,
                updated_at TIMESTAMP DEFAULT CURRENT_TIMESTAMP ON UPDATE CURRENT_TIMESTAMP
            );

File: src/backend/db/database_manager.py [hunk 2]
```diff
@@ -850,5 +866,363 @@ def delete_ticket(self, ticket_id, customer_guid):
...
+                        "SELECT updated_at, is_edited FROM ticket_comments WHERE ticket_id = :ticket_id AND comment_id = :comment_id"),
+                    {"ticket_id": ticket_id, "comment_id": comment_id}
+                ).fetchone()
+
+                session.commit()
+
+                updated_at = updated_at_result[0] if updated_at_result else None
+                is_edited = updated_at_result[1] if updated_at_result else False
+
+                return {"status": "updated", "reason": None, "comment": comment_update.comment, "ticket_id": ticket_id, "comment_id":comment_id}
```
Comment (SaravanakumarR2018) [inline src/backend/db/database_manager.py:1133]: CREATE TABLE IF NOT EXISTS ticket_comments (
comment_id BIGINT AUTO_INCREMENT PRIMARY KEY,
ticket_id BIGINT NOT NULL,
posted_by VARCHAR(255) NOT NULL,
comment TEXT NOT NULL,
is_edited BOOLEAN DEFAULT FALSE,
created_at TIMESTAMP DEFAULT CURRENT_TIMESTAMP,
updated_at TIMESTAMP DEFAULT CURRENT_TIMESTAMP ON UPDATE CURRENT_TIMESTAMP
);

File: src/backend/ticket_service/ticket_service.py [hunk 3]
```diff
@@ -376,4 +377,262 @@ async def delete_ticket(ticket_id: str, customer_guid: str):
         raise HTTPException(
             status_code=HTTPStatus.INTERNAL_SERVER_ERROR,
             detail="An unexpected error occurred while deleting the ticket."
+        )
+
+
```
Comment (SaravanakumarR2018) [inline src/backend/ticket_service/ticket_service.py:382]: Lets have one white space and remove too many white spaces

File: src/backend/ticket_service/ticket_service.py [hunk 4]
```diff
@@ -376,4 +377,262 @@ async def delete_ticket(ticket_id: str, customer_guid: str):
...
+
+#Comments APIS
+
+class Comment(BaseModel):
```
Comment (SaravanakumarR2018) [inline src/backend/ticket_service/ticket_service.py:386]: Move all the classes to top of the file

File: src/backend/db/database_manager.py [hunk 5]
```diff
@@ -850,5 +866,363 @@ def delete_ticket(self, ticket_id, customer_guid):
...
+                text(comment_query),
+                {"comment_id": comment_id, "ticket_id": ticket_id},
+            ).fetchone()
+
+            if not comment_result:
+                return None  # If comment is not found, return None
+
+            # Convert comment result into a dictionary
+            logger.info(f"Comment details by comment_id {comment_result}")
+            comment_data = {
```
Comment (SaravanakumarR2018) [inline src/backend/db/database_manager.py:1003]: The output should have

CREATE TABLE IF NOT EXISTS ticket_comments (
comment_id BIGINT AUTO_INCREMENT PRIMARY KEY,
ticket_id BIGINT NOT NULL,
posted_by VARCHAR(255) NOT NULL,
comment TEXT NOT NULL,
is_edited BOOLEAN DEFAULT FALSE,
created_at TIMESTAMP DEFAULT CURRENT_TIMESTAMP,
updated_at TIMESTAMP DEFAULT CURRENT_TIMESTAMP ON UPDATE CURRENT_TIMESTAMP
);

File: src/backend/db/database_manager.py [hunk 6]
```diff
@@ -850,5 +866,363 @@ def delete_ticket(self, ticket_id, customer_guid):
...
+                if result:
+                    comment_id = result[0]
+                    logger.debug(f"Created comment with ID: {comment_id}")
+                else:
+                    raise Exception("Failed to retrieve the newly created comment ID.")
+
+            session.commit()
+            logger.debug(f"Transaction committed successfully for comment ID {comment_id}")
+
+            return {"comment_id": comment_id, "comment": comment}
```
Comment (SaravanakumarR2018) [inline src/backend/db/database_manager.py:949]: The output should have

CREATE TABLE IF NOT EXISTS ticket_comments (
comment_id BIGINT AUTO_INCREMENT PRIMARY KEY,
ticket_id BIGINT NOT NULL,
posted_by VARCHAR(255) NOT NULL,
comment TEXT NOT NULL,
is_edited BOOLEAN DEFAULT FALSE,
created_at TIMESTAMP DEFAULT CURRENT_TIMESTAMP,
updated_at TIMESTAMP DEFAULT CURRENT_TIMESTAMP ON UPDATE CURRENT_TIMESTAMP
);

File: src/backend/ticket_service/ticket_service.py [hunk 7]
```diff
@@ -376,4 +377,262 @@ async def delete_ticket(ticket_id: str, customer_guid: str):
...
+            logger.error(f"Database error: {e}")
+            raise HTTPException(
+                status_code=HTTPStatus.SERVICE_UNAVAILABLE,
+                detail="The database is currently unreachable. Please try again later."
+            )
+        logger.error(f"Unexpected error: {e}")
+        raise HTTPException(status_code=HTTPStatus.INTERNAL_SERVER_ERROR, detail="Internal Server Error")
+
+
+@app.get("/tickets/{ticket_id}/comments", response_model=List[CommentByTicketId], tags=["Comment Management"])
```
Comment (SaravanakumarR2018) [inline src/backend/ticket_service/ticket_service.py:495]: How are you giving the inputs of page and page_size?

it should be like this
/tickets/{ticket_id}/comments?page=2&page_size=40

Refer to get all chats

Here you have only given default values as 1 and 10, but what about other pages and other page sizes?

**Review by SaravanakumarR2018 — COMMENTED** [review]
Good work!

Whenever we are returning comments, lets make sure we get a uniform output with the below fields so that there is uniformity across all APIs

CREATE TABLE IF NOT EXISTS ticket_comments (
comment_id BIGINT AUTO_INCREMENT PRIMARY KEY,
ticket_id BIGINT NOT NULL,
posted_by VARCHAR(255) NOT NULL,
comment TEXT NOT NULL,
is_edited BOOLEAN DEFAULT FALSE,
created_at TIMESTAMP DEFAULT CURRENT_TIMESTAMP,
updated_at TIMESTAMP DEFAULT CURRENT_TIMESTAMP ON UPDATE CURRENT_TIMESTAMP
);

lets make sure the uniformity is maintained in the output for the comments returning APIs

File: src/backend/db/database_manager.py (hunk 1 above)
Comment (Bharani0012) [reply src/backend/db/database_manager.py:1055]: modified sir

File: src/backend/db/database_manager.py (hunk 6 above)
Comment (Bharani0012) [reply src/backend/db/database_manager.py:949]: modified sir

File: src/backend/db/database_manager.py (hunk 5 above)
Comment (Bharani0012) [reply src/backend/db/database_manager.py:1003]: modified sir

File: src/backend/ticket_service/ticket_service.py (hunk 7 above)
Comment (Bharani0012) [reply src/backend/ticket_service/ticket_service.py:495]: Sir, the request URL is the same as mentioned above

File: src/backend/ticket_service/ticket_service.py (hunk 4 above)
Comment (Bharani0012) [reply src/backend/ticket_service/ticket_service.py:386]: Modified sir

File: src/backend/ticket_service/ticket_service.py (hunk 3 above)
Comment (Bharani0012) [reply src/backend/ticket_service/ticket_service.py:382]: modified sir

File: src/backend/db/database_manager.py (hunk 2 above)
Comment (Bharani0012) [reply src/backend/db/database_manager.py:1133]: modified sir

**Pull request closed without merging**